backend/src/tools/prompt.py

```
@@ -0,0 +1,51 @@
+    kind: Literal["prompt_memory", "gradient", "metaprompt"] = Field(default="prompt_memory", description="The kind of optimizer to use")
+    config: Optional[dict] = Field(default={"min_reflection_steps": 2, "max_reflection_steps": 3}, description="The config to use for optimization")
+    model: Optional[ChatModels] = Field(default=ChatModels.OPENAI_GPT_5_NANO, description="The model to use for optimization")
+
+
+@tool(args_schema=PromptOptimizerToolInput)
+async def prompt_optimize(
+    prompt: Prompt,
+    trajectories: List[Trajectory] | str,
+    kind: Literal["prompt_memory", "gradient", "metaprompt"] = "prompt_memory",
+    config: dict = {"min_reflection_steps": 2, "max_reflection_steps": 3},
+    model: Optional[ChatModels] = Field(default=ChatModels.OPENAI_GPT_5_NANO, description="The model to use for optimization")
```

Ruff (0.14.1)

Do not perform function call `Field` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)